Compiler error when building CAF 0.17.4 with C++17

I have compiled the project with the configure script instructions and I am getting the following error below when trying to compile the code with g++ in the examples folder specifically anything that includes caf/all.hpp.
Here is the error:
In file included from /usr/local/include/caf/message_id.hpp:27:0,
                 from /usr/local/include/caf/abstract_channel.hpp:25,
                 from /usr/local/include/caf/abstract_actor.hpp:30,
                 from /usr/local/include/caf/all.hpp:23,
                 from hello_world.cpp:4:
/usr/local/include/caf/detail/type_traits.hpp:295:8: error: redefinition of ‘struct caf::detail::callable_trait<R (C::*)(Ts ...) const noexcept>’
 struct callable_trait<R (C::*)(Ts...) const> : callable_trait<R (Ts...)> {};
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/caf/detail/type_traits.hpp:286:8: note: previous definition of ‘struct caf::detail::callable_trait<R (C::*)(Ts ...) const noexcept>’
 struct callable_trait<R (C::*)(Ts...) const noexcept>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/caf/detail/type_traits.hpp:299:8: error: redefinition of ‘struct caf::detail::callable_trait<R (C::*)(Ts ...) noexcept>’
 struct callable_trait<R (C::*)(Ts...)> : callable_trait<R (Ts...)> {};
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/caf/detail/type_traits.hpp:291:8: note: previous definition of ‘struct caf::detail::callable_trait<R (C::*)(Ts ...) noexcept>’
 struct callable_trait<R (C::*)(Ts...) noexcept> : callable_trait<R(Ts...)> {};
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/caf/detail/type_traits.hpp:307:8: error: redefinition of ‘struct caf::detail::callable_trait<R (*)(Ts ...) noexcept>’
 struct callable_trait<R (*)(Ts...)> : callable_trait<R (Ts...)> {};
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/caf/detail/type_traits.hpp:303:8: note: previous definition of ‘struct caf::detail::callable_trait<R (*)(Ts ...) noexcept>’
 struct callable_trait<R (*)(Ts...) noexcept> : callable_trait<R(Ts...)> {};

Now running the following:
`g++ -std=c++17 hello_world.cpp`
gives me this error:
/tmp/ccV7j3fv.o: In function `mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}::operator()(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
hello_world.cpp:(.text+0xdc): undefined reference to `caf::aout(caf::local_actor*)'
hello_world.cpp:(.text+0x10c): undefined reference to `std::endl(caf::actor_ostream&)'
/tmp/ccV7j3fv.o: In function `hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}::operator()(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
hello_world.cpp:(.text+0x24a): undefined reference to `caf::aout(caf::local_actor*)'
hello_world.cpp:(.text+0x27a): undefined reference to `std::endl(caf::actor_ostream&)'
/tmp/ccV7j3fv.o: In function `main':
hello_world.cpp:(.text+0x43b): undefined reference to `caf::core::init_global_meta_objects()'
/tmp/ccV7j3fv.o: In function `void caf::policy::single_response<caf::message>::then<caf::event_based_actor, hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}>(caf::event_based_actor*, caf::actor_traits<caf::event_based_actor>&&, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}&&) const':
hello_world.cpp:(.text+0x970): undefined reference to `caf::scheduled_actor::add_multiplexed_response_handler(caf::message_id, caf::behavior)'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>, caf::detail::dummy_timeout_definition>::default_behavior_impl(mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}&&, std::tuple<mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>)':
hello_world.cpp:(.text+0xa28): undefined reference to `caf::detail::behavior_impl::behavior_impl(std::chrono::duration<long, std::ratio<1l, 1000000000l> >)'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}>, caf::detail::dummy_timeout_definition>::default_behavior_impl({lambda(caf::error&)#1}&&, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1})':
hello_world.cpp:(.text+0x1098): undefined reference to `caf::detail::behavior_impl::behavior_impl(std::chrono::duration<long, std::ratio<1l, 1000000000l> >)'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}>, caf::detail::dummy_timeout_definition>::~default_behavior_impl()':
hello_world.cpp:(.text+0x1252): undefined reference to `caf::detail::behavior_impl::~behavior_impl()'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>, caf::detail::dummy_timeout_definition>::~default_behavior_impl()':
hello_world.cpp:(.text+0x12a8): undefined reference to `caf::detail::behavior_impl::~behavior_impl()'
/tmp/ccV7j3fv.o:(.data.rel.ro+0x18): undefined reference to `caf::memory_managed::request_deletion(bool) const'
/tmp/ccV7j3fv.o:(.data.rel.ro+0x50): undefined reference to `caf::memory_managed::request_deletion(bool) const'
/tmp/ccV7j3fv.o:(.data.rel.ro+0x88): undefined reference to `typeinfo for caf::detail::behavior_impl'
/tmp/ccV7j3fv.o:(.data.rel.ro+0xa0): undefined reference to `typeinfo for caf::detail::behavior_impl'
/tmp/ccV7j3fv.o: In function `caf::detail::message_data::deref()':
hello_world.cpp:(.text._ZN3caf6detail12message_data5derefEv[_ZN3caf6detail12message_data5derefEv]+0x61): undefined reference to `caf::detail::message_data::~message_data()'
/tmp/ccV7j3fv.o: In function `caf::intrusive_ptr_release(caf::ref_counted const*)':
hello_world.cpp:(.text._ZN3caf21intrusive_ptr_releaseEPKNS_11ref_countedE[_ZN3caf21intrusive_ptr_releaseEPKNS_11ref_countedE]+0x14): undefined reference to `caf::ref_counted::deref() const'
/tmp/ccV7j3fv.o: In function `caf::actor_ostream::operator<<(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
hello_world.cpp:(.text._ZN3caf13actor_ostreamlsENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN3caf13actor_ostreamlsENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x4d): undefined reference to `caf::actor_ostream::write(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
/tmp/ccV7j3fv.o: In function `auto caf::mixin::requester<caf::mixin::sender<caf::scheduled_actor, caf::event_based_actor>, caf::event_based_actor>::request<(caf::message_priority)1, long, std::ratio<1l, 1l>, caf::actor, char const (&) [13]>(caf::actor const&, std::chrono::duration<long, std::ratio<1l, 1l> >, char const (&) [13])':
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x3f): undefined reference to `caf::local_actor::new_request_id(caf::message_priority)'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x8d): undefined reference to `caf::abstract_actor::ctrl() const'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0xf2): undefined reference to `caf::local_actor::request_response_timeout(std::chrono::duration<long, std::ratio<1l, 1000000000l> >, caf::message_id)'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x127): undefined reference to `caf::abstract_actor::ctrl() const'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x17e): undefined reference to `caf::error::~error()'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x1f8): undefined reference to `caf::error::~error()'
/tmp/ccV7j3fv.o: In function `caf::intrusive_ptr<caf::actor_control_block>::~intrusive_ptr()':
hello_world.cpp:(.text._ZN3caf13intrusive_ptrINS_19actor_control_blockEED2Ev[_ZN3caf13intrusive_ptrINS_19actor_control_blockEED5Ev]+0x23): undefined reference to `caf::intrusive_ptr_release(caf::actor_control_block*)'
/tmp/ccV7j3fv.o: In function `caf::error caf::make_error<caf::sec>(caf::sec)':
hello_world.cpp:(.text._ZN3caf10make_errorINS_3secEEENS_5errorET_[_ZN3caf10make_errorINS_3secEEENS_5errorET_]+0x24): undefined reference to `caf::error::error(unsigned char, unsigned char)'
/tmp/ccV7j3fv.o: In function `caf::infer_handle_from_fun<caf::behavior (*)(caf::event_based_actor*), caf::detail::get_callable_trait<caf::behavior (*)(caf::event_based_actor*)>::type>::type caf::actor_system::spawn<(caf::spawn_options)0, caf::behavior (*)(caf::event_based_actor*)>(caf::behavior (*)(caf::event_based_actor*))':
hello_world.cpp:(.text._ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFNS_8behaviorEPNS_17event_based_actorEEJEEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitIS9_E4typeEE4typeES9_DpOT1_[_ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFNS_8behaviorEPNS_17event_based_actorEEJEEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitIS9_E4typeEE4typeES9_DpOT1_]+0x42): undefined reference to `caf::actor_config::actor_config(caf::execution_unit*, caf::local_actor*)'
/tmp/ccV7j3fv.o: In function `caf::infer_handle_from_fun<void (*)(caf::event_based_actor*, caf::actor const&), caf::detail::get_callable_trait<void (*)(caf::event_based_actor*, caf::actor const&)>::type>::type caf::actor_system::spawn<(caf::spawn_options)0, void (*)(caf::event_based_actor*, caf::actor const&), caf::actor&>(void (*)(caf::event_based_actor*, caf::actor const&), caf::actor&)':
hello_world.cpp:(.text._ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFvPNS_17event_based_actorERKNS_5actorEEJRS5_EEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitISC_E4typeEE4typeESC_DpOT1_[_ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFvPNS_17event_based_actorERKNS_5actorEEJRS5_EEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitISC_E4typeEE4typeESC_DpOT1_]+0x46): undefined reference to `caf::actor_config::actor_config(caf::execution_unit*, caf::local_actor*)'
/tmp/ccV7j3fv.o: In function `int caf::exec_main<, void (*)(caf::actor_system&)>(void (*)(caf::actor_system&), int, char**, char const*)':
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x43): undefined reference to `caf::actor_system_config::actor_system_config()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x70): undefined reference to `caf::actor_system_config::parse(int, char**, char const*)'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0xb2): undefined reference to `caf::actor_system_config::render[abi:cxx11](caf::error const&)'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x10a): undefined reference to `caf::error::~error()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x143): undefined reference to `caf::actor_system::actor_system(caf::actor_system_config&)'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x1dc): undefined reference to `caf::actor_system::~actor_system()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x1eb): undefined reference to `caf::actor_system_config::~actor_system_config()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x224): undefined reference to `caf::error::~error()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x238): undefined reference to `caf::actor_system::~actor_system()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x24c): undefined reference to `caf::actor_system_config::~actor_system_config()'
/tmp/ccV7j3fv.o: In function `std::enable_if<(!std::is_same<std::decay<char const (&) [13]>::type, caf::message>::value)||(((0)>(0))), std::unique_ptr<caf::mailbox_element, std::default_delete<caf::mailbox_element> > >::type caf::make_mailbox_element<char const (&) [13]>(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, char const (&) [13])':
hello_world.cpp:(.text._ZN3caf20make_mailbox_elementIRA13_KcJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISD_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISL_SaISL_EEOS7_DpOT0_[_ZN3caf20make_mailbox_elementIRA13_KcJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISD_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISL_SaISL_EEOS7_DpOT0_]+0xa1): undefined reference to `caf::make_mailbox_element(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, caf::message)'
/tmp/ccV7j3fv.o: In function `std::enable_if<(!std::is_same<std::decay<caf::error>::type, caf::message>::value)||(((0)>(0))), std::unique_ptr<caf::mailbox_element, std::default_delete<caf::mailbox_element> > >::type caf::make_mailbox_element<caf::error>(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, caf::error&&)':
hello_world.cpp:(.text._ZN3caf20make_mailbox_elementINS_5errorEJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISB_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISJ_SaISJ_EEOS5_DpOT0_[_ZN3caf20make_mailbox_elementINS_5errorEJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISB_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISJ_SaISJ_EEOS5_DpOT0_]+0xa1): undefined reference to `caf::make_mailbox_element(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, caf::message)'
/tmp/ccV7j3fv.o: In function `caf::detail::message_data* caf::default_intrusive_cow_ptr_unshare<caf::detail::message_data>(caf::detail::message_data*&)':
hello_world.cpp:(.text._ZN3caf33default_intrusive_cow_ptr_unshareINS_6detail12message_dataEEEPT_RS4_[_ZN3caf33default_intrusive_cow_ptr_unshareINS_6detail12message_dataEEEPT_RS4_]+0x2d): undefined reference to `caf::detail::message_data::copy() const'
/tmp/ccV7j3fv.o: In function `caf::message caf::make_message<char const (&) [13]>(char const (&) [13])':
hello_world.cpp:(.text._ZN3caf12make_messageIJRA13_KcEEENS_7messageEDpOT_[_ZN3caf12make_messageIJRA13_KcEEENS_7messageEDpOT_]+0x97): undefined reference to `caf::detail::message_data::message_data(caf::type_id_list)'
/tmp/ccV7j3fv.o: In function `caf::message caf::make_message<caf::error>(caf::error&&)':
hello_world.cpp:(.text._ZN3caf12make_messageIJNS_5errorEEEENS_7messageEDpOT_[_ZN3caf12make_messageIJNS_5errorEEEENS_7messageEDpOT_]+0x97): undefined reference to `caf::detail::message_data::message_data(caf::type_id_list)'
/tmp/ccV7j3fv.o: In function `caf::infer_handle_from_class<caf::event_based_actor, std::is_base_of<caf::abstract_actor, caf::event_based_actor>::value>::type caf::actor_system::spawn_impl<caf::event_based_actor, (caf::spawn_options)0>(caf::actor_config&)':
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0x90): undefined reference to `caf::actor_system::dummy_execution_unit()'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xa6): undefined reference to `caf::logger::set_current_actor_system(caf::actor_system*)'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xb2): undefined reference to `caf::actor_system::node() const'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xd0): undefined reference to `caf::actor_system::next_actor_id()'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xfd): undefined reference to `caf::node_id::~node_id()'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0x15d): undefined reference to `caf::node_id::~node_id()'
/tmp/ccV7j3fv.o: In function `void caf::detail::message_data_init<caf::error>(caf::byte*, caf::error&&)':
hello_world.cpp:(.text._ZN3caf6detail17message_data_initINS_5errorEJEEEvPNS_4byteEOT_DpOT0_[_ZN3caf6detail17message_data_initINS_5errorEJEEEvPNS_4byteEOT_DpOT0_]+0x38): undefined reference to `caf::error::error(caf::error&&)'
/tmp/ccV7j3fv.o: In function `caf::actor caf::make_actor<caf::event_based_actor, caf::actor, caf::actor_config&>(unsigned long, caf::node_id, caf::actor_system*, caf::actor_config&)':
hello_world.cpp:(.text._ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_[_ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_]+0x96): undefined reference to `caf::node_id::~node_id()'
hello_world.cpp:(.text._ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_[_ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_]+0xae): undefined reference to `caf::actor::actor(caf::actor_control_block*, bool)'
hello_world.cpp:(.text._ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_[_ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_]+0xe7): undefined reference to `caf::node_id::~node_id()'
/tmp/ccV7j3fv.o: In function `caf::actor_control_block::~actor_control_block()':
hello_world.cpp:(.text._ZN3caf19actor_control_blockD2Ev[_ZN3caf19actor_control_blockD5Ev]+0x18): undefined reference to `caf::node_id::~node_id()'
/tmp/ccV7j3fv.o: In function `caf::actor_storage<caf::event_based_actor>::actor_storage<caf::actor_config&>(unsigned long, caf::node_id, caf::actor_system*, caf::actor_config&)':
hello_world.cpp:(.text._ZN3caf13actor_storageINS_17event_based_actorEEC2IJRNS_12actor_configEEEEmNS_7node_idEPNS_12actor_systemEDpOT_[_ZN3caf13actor_storageINS_17event_based_actorEEC5IJRNS_12actor_configEEEEmNS_7node_idEPNS_12actor_systemEDpOT_]+0x75): undefined reference to `caf::event_based_actor::event_based_actor(caf::actor_config&)'
/tmp/ccV7j3fv.o: In function `caf::message caf::make_message<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)':
hello_world.cpp:(.text._ZN3caf12make_messageIJNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEENS_7messageEDpOT_[_ZN3caf12make_messageIJNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEENS_7messageEDpOT_]+0x97): undefined reference to `caf::detail::message_data::message_data(caf::type_id_list)'
collect2: error: ld returned 1 exit status

Now running the following:
`g++ -std=c++17 hello_world.cpp`
gives me this error:
/tmp/ccV7j3fv.o: In function `mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}::operator()(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
hello_world.cpp:(.text+0xdc): undefined reference to `caf::aout(caf::local_actor*)'
hello_world.cpp:(.text+0x10c): undefined reference to `std::endl(caf::actor_ostream&)'
/tmp/ccV7j3fv.o: In function `hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}::operator()(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
hello_world.cpp:(.text+0x24a): undefined reference to `caf::aout(caf::local_actor*)'
hello_world.cpp:(.text+0x27a): undefined reference to `std::endl(caf::actor_ostream&)'
/tmp/ccV7j3fv.o: In function `main':
hello_world.cpp:(.text+0x43b): undefined reference to `caf::core::init_global_meta_objects()'
/tmp/ccV7j3fv.o: In function `void caf::policy::single_response<caf::message>::then<caf::event_based_actor, hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}>(caf::event_based_actor*, caf::actor_traits<caf::event_based_actor>&&, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}&&) const':
hello_world.cpp:(.text+0x970): undefined reference to `caf::scheduled_actor::add_multiplexed_response_handler(caf::message_id, caf::behavior)'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>, caf::detail::dummy_timeout_definition>::default_behavior_impl(mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}&&, std::tuple<mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>)':
hello_world.cpp:(.text+0xa28): undefined reference to `caf::detail::behavior_impl::behavior_impl(std::chrono::duration<long, std::ratio<1l, 1000000000l> >)'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}>, caf::detail::dummy_timeout_definition>::default_behavior_impl({lambda(caf::error&)#1}&&, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1})':
hello_world.cpp:(.text+0x1098): undefined reference to `caf::detail::behavior_impl::behavior_impl(std::chrono::duration<long, std::ratio<1l, 1000000000l> >)'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<hello_world(caf::event_based_actor*, caf::actor const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, caf::response_handle<caf::event_based_actor, caf::policy::single_response<caf::message> >::then<caf::actor_traits<caf::event_based_actor>, {lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}, void>({lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}) const::{lambda(caf::error&)#1}>, caf::detail::dummy_timeout_definition>::~default_behavior_impl()':
hello_world.cpp:(.text+0x1252): undefined reference to `caf::detail::behavior_impl::~behavior_impl()'
/tmp/ccV7j3fv.o: In function `caf::detail::default_behavior_impl<std::tuple<mirror(caf::event_based_actor*)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>, caf::detail::dummy_timeout_definition>::~default_behavior_impl()':
hello_world.cpp:(.text+0x12a8): undefined reference to `caf::detail::behavior_impl::~behavior_impl()'
/tmp/ccV7j3fv.o:(.data.rel.ro+0x18): undefined reference to `caf::memory_managed::request_deletion(bool) const'
/tmp/ccV7j3fv.o:(.data.rel.ro+0x50): undefined reference to `caf::memory_managed::request_deletion(bool) const'
/tmp/ccV7j3fv.o:(.data.rel.ro+0x88): undefined reference to `typeinfo for caf::detail::behavior_impl'
/tmp/ccV7j3fv.o:(.data.rel.ro+0xa0): undefined reference to `typeinfo for caf::detail::behavior_impl'
/tmp/ccV7j3fv.o: In function `caf::detail::message_data::deref()':
hello_world.cpp:(.text._ZN3caf6detail12message_data5derefEv[_ZN3caf6detail12message_data5derefEv]+0x61): undefined reference to `caf::detail::message_data::~message_data()'
/tmp/ccV7j3fv.o: In function `caf::intrusive_ptr_release(caf::ref_counted const*)':
hello_world.cpp:(.text._ZN3caf21intrusive_ptr_releaseEPKNS_11ref_countedE[_ZN3caf21intrusive_ptr_releaseEPKNS_11ref_countedE]+0x14): undefined reference to `caf::ref_counted::deref() const'
/tmp/ccV7j3fv.o: In function `caf::actor_ostream::operator<<(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
hello_world.cpp:(.text._ZN3caf13actor_ostreamlsENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN3caf13actor_ostreamlsENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x4d): undefined reference to `caf::actor_ostream::write(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
/tmp/ccV7j3fv.o: In function `auto caf::mixin::requester<caf::mixin::sender<caf::scheduled_actor, caf::event_based_actor>, caf::event_based_actor>::request<(caf::message_priority)1, long, std::ratio<1l, 1l>, caf::actor, char const (&) [13]>(caf::actor const&, std::chrono::duration<long, std::ratio<1l, 1l> >, char const (&) [13])':
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x3f): undefined reference to `caf::local_actor::new_request_id(caf::message_priority)'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x8d): undefined reference to `caf::abstract_actor::ctrl() const'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0xf2): undefined reference to `caf::local_actor::request_response_timeout(std::chrono::duration<long, std::ratio<1l, 1000000000l> >, caf::message_id)'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x127): undefined reference to `caf::abstract_actor::ctrl() const'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x17e): undefined reference to `caf::error::~error()'
hello_world.cpp:(.text._ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_[_ZN3caf5mixin9requesterINS0_6senderINS_15scheduled_actorENS_17event_based_actorEEES4_E7requestILNS_16message_priorityE1ElSt5ratioILl1ELl1EENS_5actorEJRA13_KcEEEDaRKT2_NSt6chrono8durationIT0_T1_EEDpOT3_]+0x1f8): undefined reference to `caf::error::~error()'
/tmp/ccV7j3fv.o: In function `caf::intrusive_ptr<caf::actor_control_block>::~intrusive_ptr()':
hello_world.cpp:(.text._ZN3caf13intrusive_ptrINS_19actor_control_blockEED2Ev[_ZN3caf13intrusive_ptrINS_19actor_control_blockEED5Ev]+0x23): undefined reference to `caf::intrusive_ptr_release(caf::actor_control_block*)'
/tmp/ccV7j3fv.o: In function `caf::error caf::make_error<caf::sec>(caf::sec)':
hello_world.cpp:(.text._ZN3caf10make_errorINS_3secEEENS_5errorET_[_ZN3caf10make_errorINS_3secEEENS_5errorET_]+0x24): undefined reference to `caf::error::error(unsigned char, unsigned char)'
/tmp/ccV7j3fv.o: In function `caf::infer_handle_from_fun<caf::behavior (*)(caf::event_based_actor*), caf::detail::get_callable_trait<caf::behavior (*)(caf::event_based_actor*)>::type>::type caf::actor_system::spawn<(caf::spawn_options)0, caf::behavior (*)(caf::event_based_actor*)>(caf::behavior (*)(caf::event_based_actor*))':
hello_world.cpp:(.text._ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFNS_8behaviorEPNS_17event_based_actorEEJEEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitIS9_E4typeEE4typeES9_DpOT1_[_ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFNS_8behaviorEPNS_17event_based_actorEEJEEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitIS9_E4typeEE4typeES9_DpOT1_]+0x42): undefined reference to `caf::actor_config::actor_config(caf::execution_unit*, caf::local_actor*)'
/tmp/ccV7j3fv.o: In function `caf::infer_handle_from_fun<void (*)(caf::event_based_actor*, caf::actor const&), caf::detail::get_callable_trait<void (*)(caf::event_based_actor*, caf::actor const&)>::type>::type caf::actor_system::spawn<(caf::spawn_options)0, void (*)(caf::event_based_actor*, caf::actor const&), caf::actor&>(void (*)(caf::event_based_actor*, caf::actor const&), caf::actor&)':
hello_world.cpp:(.text._ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFvPNS_17event_based_actorERKNS_5actorEEJRS5_EEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitISC_E4typeEE4typeESC_DpOT1_[_ZN3caf12actor_system5spawnILNS_13spawn_optionsE0EPFvPNS_17event_based_actorERKNS_5actorEEJRS5_EEENS_21infer_handle_from_funIT0_NS_6detail18get_callable_traitISC_E4typeEE4typeESC_DpOT1_]+0x46): undefined reference to `caf::actor_config::actor_config(caf::execution_unit*, caf::local_actor*)'
/tmp/ccV7j3fv.o: In function `int caf::exec_main<, void (*)(caf::actor_system&)>(void (*)(caf::actor_system&), int, char**, char const*)':
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x43): undefined reference to `caf::actor_system_config::actor_system_config()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x70): undefined reference to `caf::actor_system_config::parse(int, char**, char const*)'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0xb2): undefined reference to `caf::actor_system_config::render[abi:cxx11](caf::error const&)'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x10a): undefined reference to `caf::error::~error()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x143): undefined reference to `caf::actor_system::actor_system(caf::actor_system_config&)'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x1dc): undefined reference to `caf::actor_system::~actor_system()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x1eb): undefined reference to `caf::actor_system_config::~actor_system_config()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x224): undefined reference to `caf::error::~error()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x238): undefined reference to `caf::actor_system::~actor_system()'
hello_world.cpp:(.text._ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc[_ZN3caf9exec_mainIJEPFvRNS_12actor_systemEEEEiT0_iPPcPKc]+0x24c): undefined reference to `caf::actor_system_config::~actor_system_config()'
/tmp/ccV7j3fv.o: In function `std::enable_if<(!std::is_same<std::decay<char const (&) [13]>::type, caf::message>::value)||(((0)>(0))), std::unique_ptr<caf::mailbox_element, std::default_delete<caf::mailbox_element> > >::type caf::make_mailbox_element<char const (&) [13]>(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, char const (&) [13])':
hello_world.cpp:(.text._ZN3caf20make_mailbox_elementIRA13_KcJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISD_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISL_SaISL_EEOS7_DpOT0_[_ZN3caf20make_mailbox_elementIRA13_KcJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISD_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISL_SaISL_EEOS7_DpOT0_]+0xa1): undefined reference to `caf::make_mailbox_element(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, caf::message)'
/tmp/ccV7j3fv.o: In function `std::enable_if<(!std::is_same<std::decay<caf::error>::type, caf::message>::value)||(((0)>(0))), std::unique_ptr<caf::mailbox_element, std::default_delete<caf::mailbox_element> > >::type caf::make_mailbox_element<caf::error>(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, caf::error&&)':
hello_world.cpp:(.text._ZN3caf20make_mailbox_elementINS_5errorEJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISB_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISJ_SaISJ_EEOS5_DpOT0_[_ZN3caf20make_mailbox_elementINS_5errorEJEEENSt9enable_ifIXoontsrSt7is_sameINSt5decayIT_E4typeENS_7messageEE5valuegtsZT0_Li0EESt10unique_ptrINS_15mailbox_elementESt14default_deleteISB_EEE4typeENS_13intrusive_ptrINS_19actor_control_blockEEENS_10message_idESt6vectorISJ_SaISJ_EEOS5_DpOT0_]+0xa1): undefined reference to `caf::make_mailbox_element(caf::intrusive_ptr<caf::actor_control_block>, caf::message_id, std::vector<caf::intrusive_ptr<caf::actor_control_block>, std::allocator<caf::intrusive_ptr<caf::actor_control_block> > >, caf::message)'
/tmp/ccV7j3fv.o: In function `caf::detail::message_data* caf::default_intrusive_cow_ptr_unshare<caf::detail::message_data>(caf::detail::message_data*&)':
hello_world.cpp:(.text._ZN3caf33default_intrusive_cow_ptr_unshareINS_6detail12message_dataEEEPT_RS4_[_ZN3caf33default_intrusive_cow_ptr_unshareINS_6detail12message_dataEEEPT_RS4_]+0x2d): undefined reference to `caf::detail::message_data::copy() const'
/tmp/ccV7j3fv.o: In function `caf::message caf::make_message<char const (&) [13]>(char const (&) [13])':
hello_world.cpp:(.text._ZN3caf12make_messageIJRA13_KcEEENS_7messageEDpOT_[_ZN3caf12make_messageIJRA13_KcEEENS_7messageEDpOT_]+0x97): undefined reference to `caf::detail::message_data::message_data(caf::type_id_list)'
/tmp/ccV7j3fv.o: In function `caf::message caf::make_message<caf::error>(caf::error&&)':
hello_world.cpp:(.text._ZN3caf12make_messageIJNS_5errorEEEENS_7messageEDpOT_[_ZN3caf12make_messageIJNS_5errorEEEENS_7messageEDpOT_]+0x97): undefined reference to `caf::detail::message_data::message_data(caf::type_id_list)'
/tmp/ccV7j3fv.o: In function `caf::infer_handle_from_class<caf::event_based_actor, std::is_base_of<caf::abstract_actor, caf::event_based_actor>::value>::type caf::actor_system::spawn_impl<caf::event_based_actor, (caf::spawn_options)0>(caf::actor_config&)':
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0x90): undefined reference to `caf::actor_system::dummy_execution_unit()'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xa6): undefined reference to `caf::logger::set_current_actor_system(caf::actor_system*)'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xb2): undefined reference to `caf::actor_system::node() const'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xd0): undefined reference to `caf::actor_system::next_actor_id()'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0xfd): undefined reference to `caf::node_id::~node_id()'
hello_world.cpp:(.text._ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_[_ZN3caf12actor_system10spawn_implINS_17event_based_actorELNS_13spawn_optionsE0EJEEENS_23infer_handle_from_classIT_XsrSt10is_base_ofINS_14abstract_actorES5_E5valueEE4typeERNS_12actor_configEDpOT1_]+0x15d): undefined reference to `caf::node_id::~node_id()'
/tmp/ccV7j3fv.o: In function `void caf::detail::message_data_init<caf::error>(caf::byte*, caf::error&&)':
hello_world.cpp:(.text._ZN3caf6detail17message_data_initINS_5errorEJEEEvPNS_4byteEOT_DpOT0_[_ZN3caf6detail17message_data_initINS_5errorEJEEEvPNS_4byteEOT_DpOT0_]+0x38): undefined reference to `caf::error::error(caf::error&&)'
/tmp/ccV7j3fv.o: In function `caf::actor caf::make_actor<caf::event_based_actor, caf::actor, caf::actor_config&>(unsigned long, caf::node_id, caf::actor_system*, caf::actor_config&)':
hello_world.cpp:(.text._ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_[_ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_]+0x96): undefined reference to `caf::node_id::~node_id()'
hello_world.cpp:(.text._ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_[_ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_]+0xae): undefined reference to `caf::actor::actor(caf::actor_control_block*, bool)'
hello_world.cpp:(.text._ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_[_ZN3caf10make_actorINS_17event_based_actorENS_5actorEJRNS_12actor_configEEEET0_mNS_7node_idEPNS_12actor_systemEDpOT1_]+0xe7): undefined reference to `caf::node_id::~node_id()'
/tmp/ccV7j3fv.o: In function `caf::actor_control_block::~actor_control_block()':
hello_world.cpp:(.text._ZN3caf19actor_control_blockD2Ev[_ZN3caf19actor_control_blockD5Ev]+0x18): undefined reference to `caf::node_id::~node_id()'
/tmp/ccV7j3fv.o: In function `caf::actor_storage<caf::event_based_actor>::actor_storage<caf::actor_config&>(unsigned long, caf::node_id, caf::actor_system*, caf::actor_config&)':
hello_world.cpp:(.text._ZN3caf13actor_storageINS_17event_based_actorEEC2IJRNS_12actor_configEEEEmNS_7node_idEPNS_12actor_systemEDpOT_[_ZN3caf13actor_storageINS_17event_based_actorEEC5IJRNS_12actor_configEEEEmNS_7node_idEPNS_12actor_systemEDpOT_]+0x75): undefined reference to `caf::event_based_actor::event_based_actor(caf::actor_config&)'
/tmp/ccV7j3fv.o: In function `caf::message caf::make_message<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)':
hello_world.cpp:(.text._ZN3caf12make_messageIJNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEENS_7messageEDpOT_[_ZN3caf12make_messageIJNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEENS_7messageEDpOT_]+0x97): undefined reference to `caf::detail::message_data::message_data(caf::type_id_list)'
collect2: error: ld returned 1 exit status

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -39,6 +39,11 @@ config = [
             extraSteps: ['coverageReport'],
         ]],
         ['Linux', [
+            builds: ['debug'],
+            tools: ['gcc8'],
+            extraFlags: ['EXTRA_FLAGS=-std=c++17']
+        ]],
+        ['Linux', [
             builds: ['release'],
             tools: ['gcc8'],
         ]],

--- a/libcaf_core/caf/detail/type_traits.hpp
+++ b/libcaf_core/caf/detail/type_traits.hpp
@@ -471,6 +471,23 @@ struct callable_trait<R (C::*)(Ts...)> : callable_trait<R (Ts...)> {};
 template <class R, class... Ts>
 struct callable_trait<R (*)(Ts...)> : callable_trait<R (Ts...)> {};
 
+#if __cplusplus >= 201703L
+
+// member const function pointer
+template <class C, typename R, class... Ts>
+struct callable_trait<R (C::*)(Ts...) const noexcept>
+  : callable_trait<R(Ts...)> {};
+
+// member function pointer
+template <class C, typename R, class... Ts>
+struct callable_trait<R (C::*)(Ts...) noexcept> : callable_trait<R(Ts...)> {};
+
+// good ol' function pointer
+template <class R, class... Ts>
+struct callable_trait<R (*)(Ts...) noexcept> : callable_trait<R(Ts...)> {};
+
+#endif
+
 template <class T>
 struct has_apply_operator {
   template <class U>

--- a/libcaf_opencl/examples/proper_matrix.cpp
+++ b/libcaf_opencl/examples/proper_matrix.cpp
@@ -27,9 +27,13 @@
 
 #include "caf/opencl/all.hpp"
 
-using namespace std;
 using namespace caf;
 using namespace caf::opencl;
+
+using std::cout;
+using std::endl;
+using std::string;
+using std::vector;
 
 using caf::detail::limited_vector;
 
@@ -111,11 +115,12 @@ private:
 
 template<size_t Size>
 string to_string(const square_matrix<Size>& m) {
-  ostringstream oss;
+  std::ostringstream oss;
   oss.fill(' ');
   for (size_t row = 0; row < Size; ++row) {
     for (size_t column = 0; column < Size; ++column)
-      oss << fixed << setprecision(2) << setw(9) << m(column, row);
+      oss << std::fixed << std::setprecision(2) << std::setw(9)
+          << m(column, row);
     oss << '\n';
   }
   return oss.str();
@@ -181,11 +186,10 @@ void multiplier(event_based_actor* self) {
 
   // send both matrices to the actor and
   // wait for results in form of a matrix_type
-  self->request(worker, chrono::seconds(5), move(m1), move(m2)).then(
-    [](const matrix_type& result) {
+  self->request(worker, std::chrono::seconds(5), std::move(m1), std::move(m2))
+    .then([](const matrix_type& result) {
       cout << "result:" << endl << to_string(result);
-    }
-  );
+    });
 }
 
 int main() {

--- a/libcaf_opencl/examples/scan.cpp
+++ b/libcaf_opencl/examples/scan.cpp
@@ -26,9 +26,14 @@
 #include "caf/all.hpp"
 #include "caf/opencl/all.hpp"
 
-using namespace std;
 using namespace caf;
 using namespace caf::opencl;
+
+using std::cerr;
+using std::cout;
+using std::endl;
+using std::string;
+using std::vector;
 
 using caf::detail::limited_vector;
 
@@ -166,8 +171,8 @@ kernel void phase_3(global uint* restrict data,
 
 } // namespace
 
-template <class T, class E = caf::detail::enable_if_t<is_integral<T>::value>>
-T round_up(T numToRound, T multiple)  {
+template <class T, class = caf::detail::enable_if_t<std::is_integral<T>::value>>
+T round_up(T numToRound, T multiple) {
   return ((numToRound + multiple - 1) / multiple) * multiple;
 }
 
@@ -180,9 +185,9 @@ int main() {
        << "' values." << endl;
   // ---- create data ----
   uvec values(problem_size);
-  random_device rd;
-  mt19937 gen(rd());
-  uniform_int_distribution<uval> val_gen(0, 1023);
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<uval> val_gen(0, 1023);
   std::generate(begin(values), end(values), [&]() { return val_gen(gen); });
   // ---- find device ----
   auto& mngr = system.opencl_manager();
@@ -205,7 +210,7 @@ int main() {
   }
   {
     // ---- general ----
-    auto dev = move(*opt);
+    auto dev = std::move(*opt);
     auto prog = mngr.create_program(kernel_source, "", dev);
     scoped_actor self{system};
     // ---- config parameters ----
@@ -243,7 +248,8 @@ int main() {
         return msg.apply([&](uref& data, uref& incs) {
           auto size = incs.size();
           range = nd_conf(size);
-          return make_message(move(data), move(incs), static_cast<uval>(size));
+          return make_message(std::move(data), std::move(incs),
+                              static_cast<uval>(size));
         });
       },
       in_out<uval,mref,mref>{},
@@ -256,7 +262,8 @@ int main() {
         return msg.apply([&](uref& data, uref& incs) {
           auto size = incs.size();
           range = nd_conf(size);
-          return make_message(move(data), move(incs), static_cast<uval>(size));
+          return make_message(std::move(data), std::move(incs),
+                              static_cast<uval>(size));
         });
       },
       in_out<uval,mref,val>{},
@@ -273,8 +280,8 @@ int main() {
         cout << " index | values |  scan  " << endl
              << "-------+--------+--------" << endl;
         for (size_t i = 0; i < problem_size; ++i)
-          cout << setw(6) << i << " | " << setw(6) << values[i] << " | "
-               << setw(6) << results[i] << endl;
+          cout << std::setw(6) << i << " | " << std::setw(6) << values[i]
+               << " | " << std::setw(6) << results[i] << std::endl;
       }
     );
   }

--- a/libcaf_opencl/test/opencl.cpp
+++ b/libcaf_opencl/test/opencl.cpp
@@ -12,9 +12,13 @@
 
 #include "caf/opencl/all.hpp"
 
-using namespace std;
 using namespace caf;
 using namespace caf::opencl;
+
+using std::cout;
+using std::endl;
+using std::string;
+using std::vector;
 
 using caf::detail::tl_at;
 using caf::detail::tl_head;
@@ -292,7 +296,7 @@ void check_vector_results(const string& description,
     cout << "Size: " << expected.size() << " vs. " << result.size() << endl;
     cout << "Differ at: " << endl;
     bool same = true;
-    for (size_t i = 0; i < min(expected.size(), result.size()); ++i) {
+    for (size_t i = 0; i < std::min(expected.size(), result.size()); ++i) {
       if (expected[i] != result[i]) {
         cout << "[" << i << "] " << expected[i] << " != " << result[i] << endl;
         same = false;
@@ -405,7 +409,7 @@ void test_opencl(actor_system& sys) {
               "semicolon is missing).");
   try {
     /* auto expected_error = */ mngr.create_program(kernel_source_error);
-  } catch (const exception& exc) {
+  } catch (const std::exception& exc) {
     CAF_MESSAGE("got: " << exc.what());
   }
 #endif // CAF_NO_EXCEPTIONS
@@ -423,7 +427,7 @@ void test_opencl(actor_system& sys) {
   );
 
   // test for manuel return size selection (max workgroup size 1d)
-  auto max_wg_size = min(dev->max_work_item_sizes()[0], size_t{512});
+  auto max_wg_size = std::min(dev->max_work_item_sizes()[0], size_t{512});
   auto reduce_buffer_size = static_cast<size_t>(max_wg_size) * 8;
   auto reduce_local_size  = static_cast<size_t>(max_wg_size);
   auto reduce_work_groups = reduce_buffer_size / reduce_local_size;
@@ -731,7 +735,7 @@ void test_in_val_out_val(actor_system& sys) {
   }, others >> wrong_msg);
 
   // test for manuel return size selection (max workgroup size 1d)
-  auto max_wg_size = min(dev->max_work_item_sizes()[0], size_t{512});
+  auto max_wg_size = std::min(dev->max_work_item_sizes()[0], size_t{512});
   auto reduce_buffer_size = static_cast<size_t>(max_wg_size) * 8;
   auto reduce_local_size  = static_cast<size_t>(max_wg_size);
   auto reduce_work_groups = reduce_buffer_size / reduce_local_size;
@@ -801,7 +805,7 @@ void test_in_val_out_mref(actor_system& sys) {
                        " (kernel passed directly)", res1, result);
   }, others >> wrong_msg);
   // test for manuel return size selection (max workgroup size 1d)
-  auto max_wg_size = min(dev->max_work_item_sizes()[0], size_t{512});
+  auto max_wg_size = std::min(dev->max_work_item_sizes()[0], size_t{512});
   auto reduce_buffer_size = static_cast<size_t>(max_wg_size) * 8;
   auto reduce_local_size  = static_cast<size_t>(max_wg_size);
   auto reduce_work_groups = reduce_buffer_size / reduce_local_size;
@@ -858,7 +862,7 @@ void test_in_mref_out_val(actor_system& sys) {
                          " (kernel passed directly)", res1, result);
   }, others >> wrong_msg);
   // test for manuel return size selection (max workgroup size 1d)
-  auto max_wg_size = min(dev->max_work_item_sizes()[0], size_t{512});
+  auto max_wg_size = std::min(dev->max_work_item_sizes()[0], size_t{512});
   auto reduce_buffer_size = static_cast<size_t>(max_wg_size) * 8;
   auto reduce_local_size  = static_cast<size_t>(max_wg_size);
   auto reduce_work_groups = reduce_buffer_size / reduce_local_size;
@@ -917,7 +921,7 @@ void test_in_mref_out_mref(actor_system& sys) {
                        " (kernel passed directly)", res1, result);
   }, others >> wrong_msg);
   // test for manuel return size selection (max workgroup size 1d)
-  auto max_wg_size = min(dev->max_work_item_sizes()[0], size_t{512});
+  auto max_wg_size = std::min(dev->max_work_item_sizes()[0], size_t{512});
   auto reduce_buffer_size = static_cast<size_t>(max_wg_size) * 8;
   auto reduce_local_size  = static_cast<size_t>(max_wg_size);
   auto reduce_work_groups = reduce_buffer_size / reduce_local_size;

